base/lock.jl

```diff
@@ -164,6 +164,17 @@ function lock(f, l::AbstractLock)
     end
 end
 
+struct Locked{L<:AbstractLock, T}
```

Comment:
Does it make sense to use `Locked{T, L<:AbstractLock}`? I'm guessing that dispatching on `Locked{<:MyType}` is more common than `Locked{<:LockType}`.

Reply:
I agree. I have changed it to `Lockable{T, L<:AbstractLock}`.